src/treetensornetworks/projttns/projttn.jl

```diff
@@ -20,38 +21,83 @@ function set_nsite(P::ProjTTN, nsite)
 end
 
 function shift_position(P::ProjTTN, pos)
-  return ProjTTN(pos, P.H, P.environments)
+  return ProjTTN(pos, operator(P), environments(P))
 end
 
-function make_environment!(P::ProjTTN{V}, psi::TTN{V}, e::NamedEdge{V})::ITensor where {V}
+set_environments(p::ProjTTN, environments) = ProjTTN(pos(p), operator(p), environments)
+function set_environment(p::ProjTTN, edge, env)
+  newenvs = merge(environments(p), Dictionary((edge,), (env,)))
+  return ProjTTN(pos(p), operator(p), newenvs)
+end
+set_environment!(p::ProjTTN, edge, env) = set!(environments(p), edge, env)
```

Comment:
```suggestion
function set_environment!(p::ProjTTN, edge, env)
  set!(environments(p), edge, env)
  return p
end
```
I like to use a convention of returning the object being modified from in-place functions.